assets/sidecars/lease_provisioner_role.yaml

```diff
@@ -0,0 +1,8 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: lease-external-provisioner-role
+rules:
+  - apiGroups: ["coordination.k8s.io"]
+    resources: ["leases"]
+    verbs: ["get", "watch", "list", "delete", "update", "create"]
```

Comment:
IMO this is not related to provisioning at all. I'd use some global-sounding role (`csi-driver-lease-leader-election`?) and use just RoleBinding (not ClusterRoleBinding), because the sidecars need these perms. only in namespaces where they run.

Reply:
> IMO this is not related to provisioning at all.

We currently have 3 operators defining this role:
https://github.com/openshift/aws-efs-csi-driver-operator/blob/master/assets/rbac/provisioner_role.yaml#L25-L26
https://github.com/openshift/azure-file-csi-driver-operator/blob/main/assets/rbac/provisioner_role.yaml#L25-L26
https://github.com/openshift/ibm-powervs-block-csi-driver-operator/blob/main/assets/rbac/provisioner_role.yaml#L31-L32